### PR TITLE
add missing known limitations

### PR DESCRIPTION
**Pull request by taroface**
During the process for identifying new 24.2 known limitations in SQL Queries, @rmloveland surfaced many that had been previously tagged but not yet documented. This PR should cover them all.

None of the known limitations turned out to be new in 24.2, so I've added them to both 24.1 and 24.2 docs.

### CODE REVIEW

**Review by DrewKimball — APPROVED** [review]
LGTM - Just had a few minor comments.

File: src/current/_includes/v24.1/known-limitations/plpgsql-limitations.md [hunk 1]
```diff
@@ -17,4 +17,10 @@
...
\ No newline at end of file
+- `OPEN FOR EXECUTE` is not supported for opening cursors. [#117744](https://github.com/cockroachdb/cockroach/issues/117744)
+- The `print_strict_params` option is not supported in PL/pgSQL. [#123671](https://github.com/cockroachdb/cockroach/issues/123671)
+- The `FOUND` local variable, which checks whether a statement affected any rows, is not supported in Pl/pgSQL. [#122306](https://github.com/cockroachdb/cockroach/issues/122306)
+- By default, when a PL/pgSQL variable conflicts with a column name, CockroachDB resolves the ambiguity by treating it as a column reference rather than a variable reference. This behavior differs from PostgreSQL, where an ambiguous column error is reported, and it is possible to change the `plpgsql.variable_conflict` setting in order to prefer either columns or variables. [#115680](https://github.com/cockroachdb/cockroach/issues/115680)
+- When coercing a PL/pgSQL expression into a type that is too narrow, CockroachDB truncates values. This differs from PostgreSQL, where PL/pgSQL type coercion does not truncate the value. [#115385](https://github.com/cockroachdb/cockroach/issues/115385)
```
Comment (DrewKimball) [inline src/current/_includes/v24.1/known-limitations/plpgsql-limitations.md:24]: Note: this is likely a Postgres bug, so maybe doesn't fit well as a CRDB "limitation". I'm not sure what the precedent is for a case like that, though.

File: src/current/_includes/v24.1/known-limitations/udf-limitations.md [hunk 2]
```diff
@@ -6,4 +6,5 @@
     - Views. [#87699](https://github.com/cockroachdb/cockroach/issues/87699)
 - User-defined functions cannot call themselves recursively. [#93049](https://github.com/cockroachdb/cockroach/issues/93049)
 - [Common table expressions]({% link {{ page.version.version }}/common-table-expressions.md %}) (CTE), recursive or non-recursive, are not supported in [user-defined functions]({% link {{ page.version.version }}/user-defined-functions.md %}) (UDF). That is, you cannot use a `WITH` clause in the body of a UDF. [#92961](https://github.com/cockroachdb/cockroach/issues/92961)
-- The `setval` function cannot be resolved when used inside UDF bodies. [#110860](https://github.com/cockroachdb/cockroach/issues/110860)
\ No newline at end of file
+- The `setval` function cannot be resolved when used inside UDF bodies. [#110860](https://github.com/cockroachdb/cockroach/issues/110860)
+- Casting subqueries to [`ENUM`]({% link {{ page.version.version }}/enum.md %})s in UDFs is not supported. [#108184](https://github.com/cockroachdb/cockroach/issues/108184)
```
Comment (DrewKimball) [inline src/current/_includes/v24.1/known-limitations/udf-limitations.md:10]: While the issue only mentions enum types, I believe this restriction applies to any user-defined type.

**Review by DrewKimball — APPROVED** [review]
Oops, forgot to flush the comments before.

**Review by rytaft — APPROVED** [review]
@DrewKimball is the expert here, but LGTM once you get his approval

File: src/current/_includes/v24.1/known-limitations/plpgsql-limitations.md (hunk 1 above)
Comment (taroface) [reply src/current/_includes/v24.1/known-limitations/plpgsql-limitations.md:24]: I agree it's a stretch. In that case, I feel we should remove the `docs-known-limitation` label from #115680 and not document it here. I'll go ahead and remove it from this PR.

File: src/current/_includes/v24.2/known-limitations/plpgsql-limitations.md [hunk 3]
```diff
@@ -17,4 +17,9 @@
 - `NOT NULL` variable declarations are not supported. [#105243](https://github.com/cockroachdb/cockroach/issues/105243)
 - Cursors opened in PL/pgSQL execute their queries on opening, affecting performance and resource usage. [#111479](https://github.com/cockroachdb/cockroach/issues/111479)
 - Cursors in PL/pgSQL cannot be declared with arguments. [#117746](https://github.com/cockroachdb/cockroach/issues/117746)
-- `OPEN FOR EXECUTE` is not supported for opening cursors. [#117744](https://github.com/cockroachdb/cockroach/issues/117744)
\ No newline at end of file
+- `OPEN FOR EXECUTE` is not supported for opening cursors. [#117744](https://github.com/cockroachdb/cockroach/issues/117744)
+- The `print_strict_params` option is not supported in PL/pgSQL. [#123671](https://github.com/cockroachdb/cockroach/issues/123671)
+- The `FOUND` local variable, which checks whether a statement affected any rows, is not supported in Pl/pgSQL. [#122306](https://github.com/cockroachdb/cockroach/issues/122306)
```
Comment (florence-crl) [inline src/current/_includes/v24.2/known-limitations/plpgsql-limitations.md:22]: nit
```suggestion
- The `FOUND` local variable, which checks whether a statement affected any rows, is not supported in PL/pgSQL. [#122306](https://github.com/cockroachdb/cockroach/issues/122306)
```

File: src/current/_includes/v24.2/known-limitations/routine-limitations.md [hunk 4]
```diff
@@ -4,4 +4,7 @@
 - Routines cannot be created if they reference temporary tables. [#121375](https://github.com/cockroachdb/cockroach/issues/121375)
 - Routines cannot be created with unnamed `INOUT` parameters. For example, `CREATE PROCEDURE p(INOUT INT) AS $$ BEGIN NULL; END; $$ LANGUAGE PLpgSQL;`. [#121251](https://github.com/cockroachdb/cockroach/issues/121251)
 - Routines cannot be created if they return fewer columns than declared. For example, `CREATE FUNCTION f(OUT sum INT, INOUT a INT, INOUT b INT) LANGUAGE SQL AS $$ SELECT (a + b, b); $$;`. [#121247](https://github.com/cockroachdb/cockroach/issues/121247)
-- DDL statements (e.g., `CREATE TABLE`, `CREATE INDEX`) are not allowed within UDFs or stored procedures. [#110080](https://github.com/cockroachdb/cockroach/issues/110080)
\ No newline at end of file
+- Routines cannot be created with an `OUT` parameter of type `RECORD`. [#123448](https://github.com/cockroachdb/cockroach/issues/123448)
```
Comment (florence-crl) [inline src/current/_includes/v24.2/known-limitations/routine-limitations.md:7]: btw, I could not find public docs for type `RECORD`.

**Review by florence-crl — APPROVED** [review]
lgtm with 1 non-blocking nit

File: src/current/_includes/v24.2/known-limitations/routine-limitations.md (hunk 4 above)
Comment (taroface) [reply src/current/_includes/v24.2/known-limitations/routine-limitations.md:7]: @DrewKimball Is this specific to routines, or are we actually missing a `RECORD` data type page (like https://www.cockroachlabs.com/docs/stable/int etc.)?

**Pull request merged**